torch/distributed/rpc/backend_registry.py

```diff
@@ -267,6 +273,56 @@ def _tensorpipe_check_local_device_maps(name, device_count, options):
             f"device_maps = {options.device_maps},\n"
             f"devices = {options.devices}"
         )
+    return local_devices
+
+def _update_group_membership(worker_info, my_devices, reverse_device_map):
+    agent = api._get_current_rpc_agent()
+    ret = agent._update_group_membership(worker_info, my_devices, reverse_device_map, True)
+    return "hi"
+
+def _get_device_infos():
+    if torch.cuda.is_available():
+        device_count = torch.cuda.device_count()
+    else:
+        device_count = 0
+    agent = api._get_current_rpc_agent()
+    opts = agent._get_backend_options()
+    if torch.cuda.is_available():
+        device_count = torch.cuda.device_count()
+    else:
+        device_count = 0
+    return device_count, opts.device_maps, opts.devices
+
+def _set_devices_and_reverse_device_map(agent, my_rank, my_device_maps):
```

Comment:
`my_device_maps` not used in this function? 

Just wanna make sure I understand the logic correctly. If `my_device_maps` contains a worker X that's not joined yet, that part of my device map will be activated and verified when that future worker X calls this method?

Reply:
Removed `my_device_maps`.

>Just wanna make sure I understand the logic correctly. If my_device_maps contains a worker X that's not joined yet, that part of my device map will be activated and verified when that future worker X calls this method?

Yes, is an existing worker has a worker not yet joined but in it's device map (in your example "worker X"), the devices of worker X is not yet checked. When worker X joins, the device map of itself and the workers targeting it are validated.